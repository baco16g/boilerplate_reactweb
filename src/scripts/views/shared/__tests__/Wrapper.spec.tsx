import * as React from 'react';
import {render, cleanup, fireEvent} from 'react-testing-library';
import 'jest-dom/extend-expect';
import 'jest-styled-components';
import Wrapper, {STORAGE_KEY as COLOR_MODE_STORAGE_KEY} from '../../shared/Wrapper';
import {AppProvider} from '../../../providers/AppProvider';

afterEach(cleanup);

it('matches snapshot', () => {
  const tree = (
    <AppProvider>
      <Wrapper>
        <p>this is a dummy content in wrapper</p>
      </Wrapper>
    </AppProvider>
  );
  const {asFragment} = render(tree);
  expect(asFragment()).toMatchSnapshot();
});

describe('ColorMode', () => {
  it('should save a color mode to sessionStorage', () => {
    const tree = (
      <AppProvider>
        <Wrapper>
          <p>this is a dummy content in wrapper</p>
        </Wrapper>
      </AppProvider>
    );
    const {getByText} = render(tree);
    fireEvent.click(getByText('Change Color Mode!'));
    expect(sessionStorage.__STORE__[COLOR_MODE_STORAGE_KEY]).toBe(
      JSON.stringify({colorMode: 'DARK'})
    );
    fireEvent.click(getByText('Change Color Mode!'));
    expect(sessionStorage.__STORE__[COLOR_MODE_STORAGE_KEY]).toBe(
      JSON.stringify({colorMode: 'LIGHT'})
    );
  });
});
