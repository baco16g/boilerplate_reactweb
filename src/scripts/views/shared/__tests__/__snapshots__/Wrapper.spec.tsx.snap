// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  .c2 {
  display: inline-block;
  padding: 0.3em 1em;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 12px;
  color: #67c5ff;
  border: solid 2px #67c5ff;
  border-radius: 3px;
  -webkit-transition: .4s;
  transition: .4s;
}

.c2:hover {
  background: #67c5ff;
  color: white;
}

.c0 {
  font-family: Futura;
  height: 100vh;
}

.c1 {
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;
  margin: auto;
  text-align: center;
  padding: 12px;
  box-sizing: border-box;
  background: #fff;
}

<div
    class="c0"
  >
    <p>
      this is a dummy content in wrapper
    </p>
    <div
      class="c1"
    >
      <button
        class="c2"
        color="#67c5ff"
      >
        Change Color Mode!
      </button>
    </div>
  </div>
</DocumentFragment>
`;
